charts/istio-alerts/templates/service-prometheusrule.yaml

```diff
@@ -1,42 +1,50 @@
-{{ $values           := .Values }}
-{{ $release          := .Release }}
-{{ if .Values.serviceRules.enabled }}
+{{ $values     := .Values }}
+{{ $release    := .Release }}
 
+{{ if .Values.serviceRules.enabled }}
 apiVersion: monitoring.coreos.com/v1
 kind: PrometheusRule
 metadata:
-  name: {{ .Release.Name }}-service-rules
+  name: {{ .Release.Name }}-{{ lower .Chart.Name }}-service-rules
   annotations:
     nextdoor.com/chart: {{ .Values.chart_name }}
     nextdoor.com/source: {{ .Values.chart_source }}
 spec:
   groups:
-  - name: {{ .Release.Name }}.{{ .Release.Namespace }}.serviceRules
+  - name: {{ .Release.Namespace }}.{{ .Release.Name }}.{{ .Chart.Name }}.IstioServiceRules
     rules:
     {{ with .Values.serviceRules.http5XXMonitor }}
     {{ if .enabled }}
-    - alert: {{ $release.Name }}-5xx-Rate-Too-High
+    - alert: 5xx-Rate-Too-High
       annotations:
         summary: {{`High rate of 5xx errors from {{$labels.destination_service_name}} in namespace {{$labels.namespace}}`}}
         runbook_url: {{ $values.defaults.runbookUrl }}#5xx-Rate-Too-High
         description: >-
           High rate of 5xx responses from the {{`$labels.destination_service_name`}} VirtualService in namespace {{`{{$labels.namespace}}`}}.
       expr: >-
-        sum by (destination_service_name) (increase(istio_requests_total{response_code=~"5.*", destination_service_namespace="{{ $release.Namespace }}"}[60s]))
-         / sum by (destination_service_name) (increase(istio_requests_total{destination_service_namespace="{{ $release.Namespace }}"}[60s])) > {{ .threshold }}
+        sum by (destination_service_name) (
+            increase(istio_requests_total{response_code=~"5.*", destination_service_namespace="{{ $release.Namespace }}"}[60s])
+        )
+
+          /
+
+        sum by (destination_service_name) (
+            increase(istio_requests_total{destination_service_namespace="{{ $release.Namespace }}"}[60s])
+        )
+          > {{ .threshold }}
       for: {{ .for }}
       labels:
         severity: {{ .severity }}
-        {{- if $values.defaults.additionalRuleLabels }}
-        {{ toYaml $values.defaults.additionalRuleLabels | nindent 8 }}
+        {{- with $values.defaults.additionalRuleLabels}}
+        {{ toYaml . | nindent 8}}
         {{- end }}
     {{- end }}
     {{- end }}
 
      # Adding latency alarm to trigger when request latency is above > threshold
     {{ with .Values.serviceRules.highRequestLatency }}
     {{ if .enabled }}
-    - alert: {{ $release.Name }}-HighRequestLatency
```

Comment:
I removed this because we don't need the prefix here to identify where the alert came from in my opinion. If we want to help explain the application that is having the alert issue to the oncall-engineer, then we should do that through the `summary` field IMO.

Reply:
I don't understand this at all. It's really helpful to _what service is being affected_ be front and center in an alert.

Reply:
Compare with the alarm name for the 5xx alarm on the ALB:

`AlarmName: {'Fn::Sub': '${AWS::StackName}-5xxCount-Too-High'}`

Reply:
So I totally agree that the developer getting paged wants as much context as possible. I just think the Alarm Name isn't necessarily the only way to do that. For example, take this PromethusRule:
```
    - alert: KubeQuotaAlmostFull
      annotations:
        description: 'Namespace {{ $labels.namespace }} is using {{ $value }} of its
          {{ $labels.resource }} quota. '
        runbook_url: https://github.com/Nextdoor/k8s-charts/blob/main/charts/prometheus-alerts/runbook.md#KubeQuotaAlmostFull
        summary: Namespace quota is going to be full.
      expr: |-
        (
          kube_resourcequota{job="kube-state-metrics", type="used"}
            / ignoring(instance, job, type)
          (kube_resourcequota{job="kube-state-metrics", type="hard"} > 0)
        ) * 100 > 90 < 100
      for: 10m
      labels:
        severity: warning
```
here's the alert that a developer sees in PagerDuty:

![Screen Shot 2022-01-12 at 10 01 14 AM](https://user-images.githubusercontent.com/768067/149196715-79e681f7-26c0-4fe1-a9ab-8c4cfe28697f.png)

So the page comes through with the Alert Name and the `summary` field. The summary field already includes the service name and namespace... but we could reformat it so it's more clear:

```
summary: {{`{{$labels.destination_service_name}} throwing a high rate of 5xx errors`}}
```